### PR TITLE
#169649552 notification status

### PR DESCRIPTION
**Pull request by urbanishimwe**
#### What does this PR do?
- Mark a notification as read or unread
#### Description of Task to be completed?
This will allow a user to mark a notification as read or unread by passing into params
#### How should this be manually tested?
to make a notification as read, you simply need to pass the details of it. inside a route itself:
PATCH:`api/v1/notification/:id/status` *e.g: /api/v1/notifications/2/status*
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[story](https://www.pivotaltracker.com/story/show/169649552)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

**Review by swaibat — APPROVED** [review]

File: src/routes/notification.route.js [hunk 1]
```diff
@@ -10,10 +11,11 @@ const { checkNotificationExists, checkNotificationOwner } = NotificationMiddlewa
 const {
   getUserNotifications, updateNotificationStatus, getSingleNotification, markAllNotificationsAsRead
 } = NotificationController;
+const { validateNotificationStatus } = Validation;
 
 app.get('/', checkToken, getUserNotifications);
 app.get('/:id', checkToken, checkNotificationExists, checkNotificationOwner, getSingleNotification);
-app.patch('/:id/:status', checkToken, checkNotificationExists, checkNotificationOwner, updateNotificationStatus);
+app.patch('/:id/:status', validateNotificationStatus, checkToken, checkNotificationExists, checkNotificationOwner, updateNotificationStatus);
```
Comment (elchroy) [inline src/routes/notification.route.js]: Use the approach of a toggle/switch. You don't need the user to pass in data to read or unread. Just one route to handle both. If `isRead` is `true`, change to `false`. If `false`, change to `true`. Check out `/readall` for reference

Comment (urbanishimwe) [reply src/routes/notification.route.js]: great

**Review by lemoissonM — CHANGES_REQUESTED** [review]
great work! It would also be great if the message is a little bit detailed. like saying status set to false or true

**Review by lemoissonM — CHANGES_REQUESTED** [review]
Also, when someone put pppp as notification ID, it says notification not found and status = 404. For the sake of clarity and convention, this should be 400 bad request

File: src/routes/notification.route.js [hunk 2]
```diff
@@ -10,10 +11,11 @@ const { checkNotificationExists, checkNotificationOwner } = NotificationMiddlewa
 const {
   getUserNotifications, updateNotificationStatus, getSingleNotification, markAllNotificationsAsRead
 } = NotificationController;
+const { idValidate } = Validation;
```
Comment (lemoissonM) [inline src/routes/notification.route.js:14]: great work! just set the name to this function to validateNotificationId it would be great

**Review by lemoissonM — CHANGES_REQUESTED** [review]

**Review by lemoissonM — APPROVED** [review]

**Pull request merged**